File: 258-add-digits/add-digits.java
class Solution {
    public int addDigits(int num) {
        int sum=0;
        int x,b,y;
        y=0;
        if(num<=9){
            return num;
        }
        while(num>9){
            b=num%10;
            x=num/10;
            sum=b+x;
            if(sum>=1&&sum<=9){
                y=sum;
                break;
            }
            else if(sum==0){
                y=sum;
                break;

            }
            else{
                num=sum;
            }
        }

 return y;
    }
}

File: 258-add-digits/add-digits.py
class Solution(object):
    def addDigits(self, num):
        sum=0
        while True:
            a=num%10
            b=num/10
            sum=a+b
            if(sum>=1 and sum<=9):
                return sum
                break
            elif(sum==0):
                return sum
                break
            else:
                num=sum